crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -752,206 +752,221 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
         visited: &mut FxHashSet<PackageName>,
         request_sink: &Sender<Request>,
     ) -> Result<Option<ResolverVersion>, ResolveError> {
-        let url = package.name().and_then(|name| fork_urls.get(name));
-        match (&**package, url) {
-            (PubGrubPackageInner::Root(_), _) => {
+        match &**package {
+            PubGrubPackageInner::Root(_) => {
                 Ok(Some(ResolverVersion::Available(MIN_VERSION.clone())))
             }
 
-            (PubGrubPackageInner::Python(_), _) => {
+            PubGrubPackageInner::Python(_) => {
                 // Dependencies on Python are only added when a package is incompatible; as such,
                 // we don't need to do anything here.
                 // we don't need to do anything here.
                 Ok(None)
             }
 
-            (
-                PubGrubPackageInner::Marker { name, .. }
-                | PubGrubPackageInner::Extra { name, .. }
-                | PubGrubPackageInner::Dev { name, .. }
-                | PubGrubPackageInner::Package { name, .. },
-                Some(url),
-            ) => {
-                debug!(
-                    "Searching for a compatible version of {package} @ {} ({range})",
-                    url.verbatim
-                );
+            PubGrubPackageInner::Marker { name, .. }
+            | PubGrubPackageInner::Extra { name, .. }
+            | PubGrubPackageInner::Dev { name, .. }
+            | PubGrubPackageInner::Package { name, .. } => {
+                if let Some(url) = package.name().and_then(|name| fork_urls.get(name)) {
+                    self.choose_version_url(name, range, url)
+                } else {
+                    self.choose_version_registry(name, range, package, pins, visited, request_sink)
+                }
+            }
+        }
+    }
 
-                let dist = PubGrubDistribution::from_url(name, url);
```

Comment:
It's the same code just dedented, see https://app.semanticdiff.com/gh/astral-sh/uv/pull/4543/changes for a better diff